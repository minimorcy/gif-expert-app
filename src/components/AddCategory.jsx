import { useState } from "react"
import { PropTypes } from "prop-types"

export const AddCategory = ({onNewCategory}) => 
{
    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()

        if(inputValue.trim().length <= 1)
            return;

        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input type="text" placeholder="Buscar gifs" value={inputValue} onChange={onInputChange} />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
