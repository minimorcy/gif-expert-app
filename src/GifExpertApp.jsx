import { useState } from 'react'
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {

    //  API_KEY: jMEWMlcs2Noqb7OTJtLpjAfwMuoZOqaD

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball'])

    const onAddCategory = (newCategory) => 
    {
        if(!categories.includes(newCategory))
            setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onNewCategory={(value) => onAddCategory(value)} />

            {
                categories.map((category) =>
                    (
                        <GifGrid key={category} category={category} />
                    )
                )
            }
        </>
    )
}
