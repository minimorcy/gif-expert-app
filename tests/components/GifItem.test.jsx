import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Pruebas para el componente <GifItem title url />', () => 
{
    const title = "Deadpool"
    const url = "https://media0.giphy.com/media/3dhlSINU1H0KMHopTJ/giphy.gif?cid=a5b6b05brw23bixg2aqv6r5bn5pa4ls2r8ueovgmu4h3y64e&ep=v1_gifs_search&rid=giphy.gif&ct=g"

    test('Componente sin las propiedades oblitatorias ', () => {
        render(<GifItem title={title} />)
    });
 
    test('Hacer match con el snapshot ', () => {
        const {container} = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    });
    
    test('Debe mostrar la imagen con el RUL y el ALT indicado', () => {
        const {container} = render(<GifItem title={title} url={url} />)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(url)

        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    });

    test('Debe de mostrar el tÃ­tulo en el componente', () => {
        const {container} = render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    });
});