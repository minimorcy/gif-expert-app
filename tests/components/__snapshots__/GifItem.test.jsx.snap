// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifItem title url /> Hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Deadpool"
    />
    <p>
      Deadpool
    </p>
  </div>
</div>
`;
